batch/batch/worker/worker.py

```diff
@@ -419,6 +419,10 @@ async def _fuse_mount(self, destination: str, *, credentials_path: str, tmp_path
     async def _fuse_unmount(self, path: str):
         assert CLOUD_WORKER_API
         await CLOUD_WORKER_API.unmount_cloudfuse(path)
+        with open('/proc/mounts', 'r') as f:
+            output = f.read()
+            if path in output:
+                raise IncompleteCloudFuseCleanup(f'incomplete cloudfuse unmounting: {output}')
```

Comment:
This would read this file for each mount rather than once after all attempts to unmount. Let's make JVMJob symmetric to DockerJob and check the mounts in `cleanup` in both Jobs. I think we should also make sure we attempt to upload the log before we check /proc/mounts.

Reply:
Why are we checking for paths in the scratch directory? Should I be checking for the scratch directory in JVM jobs as well?